### PR TITLE
Taskauth

### PR DESCRIPTION
**Pull request by mikeynap**
Authorization for Progress/Outputs from Tasker.

### CODE REVIEW

File: api/v1/server/middleware.go [hunk 1]
```diff
@@ -53,3 +52,14 @@ func (s *Server) AuthUserAdmin(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+func (s *Server) AuthTasker(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		u, ok := r.Context().Value(authUserKey).(user.User)
+		if u == nil || !ok || !(user.HasRole(u, user.ROLE_TASKER) || user.HasRole(u, user.ROLE_ADMIN)) {
```
Comment (meganokeefe) [inline api/v1/server/middleware.go]: Q. on the last part of this `if` - why treat `ROLE_ADMIN` as `unauthorized`? is it possible for a user to have both `ROLE_TASKER` and `ROLE_ADMIN`?

File: auth/tokenizer/jwt.go [hunk 2]
```diff
@@ -36,13 +37,15 @@ func NewJWT(secret string, ttl ...int64) *JWTTokenizer {
...
+}
+
+func (j JWTTokenizer) tokenize(u user.User, exp int64) (Token, error) {
 	if u == nil || u.Name() == "" {
 		return nil, ErrorNoID
 	}
 
-	exp := time.Now().Unix() + j.ttl
 	claims := jwt.StandardClaims{
 		NotBefore: time.Now().Unix() - 5,
```
Comment (meganokeefe) [inline auth/tokenizer/jwt.go]: Not part of this PR, but why ` - 5` ?

File: auth/tokenizer/null.go [hunk 3]
```diff
@@ -13,6 +13,10 @@ func NewNull() *NullTokenizer {
 	return &NullTokenizer{}
 }
 
+func (j NullTokenizer) LongLivedTokenize(u user.User) (Token, error) {
+	return authToken{u.Name(), time.Now().Add(1000000 * time.Hour)}, nil
```
Comment (meganokeefe) [inline auth/tokenizer/null.go]: NIT, `jwt` and `null` both have a different `exp` for longLived. 

aka `int64(1.577e+9)` vs. `1000000 * time.Hour`

File: auth/user/user.go [hunk 4]
```diff
@@ -13,10 +13,19 @@ const (
...
+	ROLE_TASKER
 	ROLE_ADMIN
 	ROLE_USERADMIN
 )
 
+func NewTaskerUser() User {
+	return AuthUser{
+		AuthName:           "tasker",
+		AuthRole:           ROLE_TASKER,
+		AuthHashedPassword: "###NoLoggyInny###",
```
Comment (meganokeefe) [inline auth/user/user.go]: can you change this lol

File: task/docker.go [hunk 5]
```diff
@@ -376,11 +380,7 @@ func (d *DockerTasker) pull(ctx context.Context, image string, creds string) err
 	if readErr != nil {
 		return readErr
 	}
-	re := regexp.MustCompile("\"error\"\\s?:\\s?\"(.*)\"")
-	if re == nil {
-		return errors.New(string(pullOutput))
-	}
-	errsOutput := re.FindSubmatch(pullOutput)
+	errsOutput := errorRegex.FindSubmatch(pullOutput)
```
Comment (meganokeefe) [inline task/docker.go]: Nice

File: auth/tokenizer/jwt.go (hunk 2 above)
Comment (mikeynap) [inline auth/tokenizer/jwt.go]: To accommodate small differences in clock times.

File: auth/tokenizer/null.go (hunk 3 above)
Comment (mikeynap) [inline auth/tokenizer/null.go]: yeah there's no good reason haha. I updated it.

File: api/v1/server/middleware.go (hunk 1 above)
Comment (mikeynap) [inline api/v1/server/middleware.go]: You're not demorganing it right haha.
This means it needs either ROLE_TASKER or ROLE_ADMIN.

Comment (mikeynap) [inline api/v1/server/middleware.go]: A user with role: ROLE_TASKER & ROLE_ADMIN & ROLE_USERADMIN
user.HasRole(ROLE_TASKER) = true
user.HasRole(ROLE_ADMIN) = true
...
etc.

Comment (meganokeefe) [inline api/v1/server/middleware.go]: Missed the parentheses, whoops #badeyes
